### ISSUE TITLE
N805 - method with no args shows no lint warnings

### ISSUE THREAD

**Issue by s-cork** [astral-sh/ruff#11600]
```python
class Foo:
    def baz(): # no lint warning
        return 42

    def bar(s): # correctly gives lint warning
        return 42
```

I don't see a lint warning for this method
checked in the ruff playground
I'd expect `N805: First argument of a method should be named "self"`

**Comment by charliermarsh** [comment]
This would differ from `pep8-naming`... though it's almost certainly an error.

**Comment by charliermarsh** [comment]
I gave this a try in https://github.com/astral-sh/ruff/pull/11605#issuecomment-2138575064 but decided against changing it after reviewing the ecosystem results. There are a ton of cases of folks omitting arguments from methods (especially in test suites) that seem valid. The rule is really about naming, and if you omitted the argument entirely, that's actually a correctness issue which requires different logic.

**Comment by charliermarsh** [comment]
Let me revisit after fixing one bug in that PR.

**Comment by s-cork** [comment]
It might also be that you move this to E0211 rather than N805

See this issue where E0211 is marked as implemented: https://github.com/astral-sh/ruff/issues/970
https://britonad.github.io/pylint-errors/plerr/errors/classes/E0211.html

**Comment by charliermarsh** [comment]
We may want another rule for it eventually but after reviewing the changes, I think we should leave the pep8-naming rules as they are.
